MB-5483 MB-5543 MB-5544 Update service to check for configuration vars or error

gidjin opened the pull request:
## Description

As part of MB-5483 we want to ensure that the variables for configuring Syncada SFTP connection are passed in. If they are not present we should gracefully handle that. This PR implements MB-5543 and MB-5544 to check for the presence of the configuration options and return an error if they are missing.

## Reviewer Notes

Not that I can think of

## Setup

```sh
make server_test
make db_dev_e2e_populate
make server_run
make bin/prime-api-client
```

Save this payload
```json
{
  "body": {
    "sendToSyncada": false
  }
}
```

Send above payload
```sh
❯ prime-api-client --insecure support-reviewed-payment-requests --filename tmp/payloads/local_process_payment.json | jq .
[
  {
    "eTag": "MjAyMC0xMS0yMFQxNjo0MTozNC42MzQ5Mzla",
    "id": "cfd110d4-1f62-401c-a92c-39987a0b4228",
    "isFinal": false,
    "moveTaskOrderID": "7cbe57ba-fd3a-45a7-aa9a-1970f1908ae7",
    "paymentRequestNumber": "8777-4251-1",
    "status": "SENT_TO_GEX"
  }
]
```

If you set `sendToSyncada` to true and unset one of the syncada variables like `SYNCADA_SFTP_PASSWORD` before starting the server you should see following error logged in the server log, but still get the above response

```sh
2020-11-20T16:43:35.337Z        ERROR   payment_request/payment_request_reviewed_processor.go:54        Configuration of SyncadaSFTPSession failed: Invalid credentials sftp missing SYNCADA_SFTP_PASSWORD      {"git_branch": "MB-5483_pass_variables_needed_to_connect_to_syncada_via_sftp", "git_commit": "f74b8e5933ff8d6eb49e1e2413cfc640bbd5009d", "milmove_trace_id": "68db60af-de30-4ec8-84bf-2e767b02d3ad"}
github.com/transcom/mymove/pkg/services/payment_request.InitNewPaymentRequestReviewedProcessor
        /Users/john/projects/dod/mymove/pkg/services/payment_request/payment_request_reviewed_processor.go:54
github.com/transcom/mymove/pkg/handlers/supportapi.ProcessReviewedPaymentRequestsHandler.Handle
        /Users/john/projects/dod/mymove/pkg/handlers/supportapi/payment_request.go:260
github.com/transcom/mymove/pkg/gen/supportapi/supportoperations/payment_request.(*ProcessReviewedPaymentRequests).ServeHTTP
        /Users/john/projects/dod/mymove/pkg/gen/supportapi/supportoperations/payment_request/process_reviewed_payment_requests.go:60
github.com/go-openapi/runtime/middleware.NewOperationExecutor.func1
        /Users/john/.asdf/installs/golang/1.15.2/packages/pkg/mod/github.com/go-openapi/runtime@v0.19.23/middleware/operation.go:28
net/http.HandlerFunc.ServeHTTP
        /Users/john/.asdf/installs/golang/1.15.2/go/src/net/http/server.go:2042
github.com/go-openapi/runtime/middleware.NewRouter.func1
        /Users/john/.asdf/installs/golang/1.15.2/packages/pkg/mod/github.com/go-openapi/runtime@v0.19.23/middleware/router.go:77
net/http.HandlerFunc.ServeHTTP
        /Users/john/.asdf/installs/golang/1.15.2/go/src/net/http/server.go:2042
github.com/go-openapi/runtime/middleware.Redoc.func1
        /Users/john/.asdf/installs/golang/1.15.2/packages/pkg/mod/github.com/go-openapi/runtime@v0.19.23/middleware/redoc.go:72
net/http.HandlerFunc.ServeHTTP
        /Users/john/.asdf/installs/golang/1.15.2/go/src/net/http/server.go:2042
github.com/go-openapi/runtime/middleware.Spec.func1
        /Users/john/.asdf/installs/golang/1.15.2/packages/pkg/mod/github.com/go-openapi/runtime@v0.19.23/middleware/spec.go:46
net/http.HandlerFunc.ServeHTTP
        /Users/john/.asdf/installs/golang/1.15.2/go/src/net/http/server.go:2042
goji%2eio.dispatch.ServeHTTP
        /Users/john/.asdf/installs/golang/1.15.2/packages/pkg/mod/goji.io@v2.0.2+incompatible/dispatch.go:17
github.com/felixge/httpsnoop.CaptureMetrics.func1
        /Users/john/.asdf/installs/golang/1.15.2/packages/pkg/mod/github.com/felixge/httpsnoop@v1.0.1/capture_metrics.go:30
github.com/felixge/httpsnoop.CaptureMetricsFn
        /Users/john/.asdf/installs/golang/1.15.2/packages/pkg/mod/github.com/felixge/httpsnoop@v1.0.1/capture_metrics.go:81
github.com/felixge/httpsnoop.CaptureMetrics
        /Users/john/.asdf/installs/golang/1.15.2/packages/pkg/mod/github.com/felixge/httpsnoop@v1.0.1/capture_metrics.go:29
github.com/transcom/mymove/pkg/middleware.RequestLogger.func1.1
        /Users/john/projects/dod/mymove/pkg/middleware/request_logger.go:77
net/http.HandlerFunc.ServeHTTP
        /Users/john/.asdf/installs/golang/1.15.2/go/src/net/http/server.go:2042
github.com/transcom/mymove/pkg/middleware.NoCache.func1.1
        /Users/john/projects/dod/mymove/pkg/middleware/no_cache.go:13
net/http.HandlerFunc.ServeHTTP
        /Users/john/.asdf/installs/golang/1.15.2/go/src/net/http/server.go:2042
github.com/transcom/mymove/pkg/auth/authentication.PrimeAuthorizationMiddleware.func1.1
        /Users/john/projects/dod/mymove/pkg/auth/authentication/auth.go:241
net/http.HandlerFunc.ServeHTTP
        /Users/john/.asdf/installs/golang/1.15.2/go/src/net/http/server.go:2042
github.com/transcom/mymove/pkg/auth/authentication.ClientCertMiddleware.func1.1
        /Users/john/projects/dod/mymove/pkg/auth/authentication/client_cert.go:61
net/http.HandlerFunc.ServeHTTP
        /Users/john/.asdf/installs/golang/1.15.2/go/src/net/http/server.go:2042
github.com/transcom/mymove/pkg/auth.HostnameDetectorMiddleware.func1.1
        /Users/john/projects/dod/mymove/pkg/auth/hostname_detector.go:21
net/http.HandlerFunc.ServeHTTP
        /Users/john/.asdf/installs/golang/1.15.2/go/src/net/http/server.go:2042
goji%2eio.(*Mux).ServeHTTP
        /Users/john/.asdf/installs/golang/1.15.2/packages/pkg/mod/goji.io@v2.0.2+incompatible/mux.go:74
goji%2eio.dispatch.ServeHTTP
        /Users/john/.asdf/installs/golang/1.15.2/packages/pkg/mod/goji.io@v2.0.2+incompatible/dispatch.go:17
github.com/transcom/mymove/pkg/middleware.LimitBodySize.func1.1
        /Users/john/projects/dod/mymove/pkg/middleware/limit_body_size.go:15
net/http.HandlerFunc.ServeHTTP
        /Users/john/.asdf/installs/golang/1.15.2/go/src/net/http/server.go:2042
github.com/transcom/mymove/pkg/middleware.SecurityHeaders.func1.1
        /Users/john/projects/dod/mymove/pkg/middleware/security_headers.go:28
net/http.HandlerFunc.ServeHTTP
        /Users/john/.asdf/installs/golang/1.15.2/go/src/net/http/server.go:2042
github.com/transcom/mymove/pkg/middleware.Recovery.func1.1
        /Users/john/projects/dod/mymove/pkg/middleware/recovery.go:50
net/http.HandlerFunc.ServeHTTP
        /Users/john/.asdf/installs/golang/1.15.2/go/src/net/http/server.go:2042
github.com/transcom/mymove/pkg/middleware.ContextLogger.func1.1
        /Users/john/projects/dod/mymove/pkg/middleware/context_logger.go:18
net/http.HandlerFunc.ServeHTTP
        /Users/john/.asdf/installs/golang/1.15.2/go/src/net/http/server.go:2042
github.com/transcom/mymove/pkg/middleware.Trace.func1.1
        /Users/john/projects/dod/mymove/pkg/middleware/trace.go:29
net/http.HandlerFunc.ServeHTTP
        /Users/john/.asdf/installs/golang/1.15.2/go/src/net/http/server.go:2042
goji%2eio.(*Mux).ServeHTTP
        /Users/john/.asdf/installs/golang/1.15.2/packages/pkg/mod/goji.io@v2.0.2+incompatible/mux.go:74
goji%2eio.dispatch.ServeHTTP
        /Users/john/.asdf/installs/golang/1.15.2/packages/pkg/mod/goji.io@v2.0.2+incompatible/dispatch.go:17
goji%2eio.(*Mux).ServeHTTP
        /Users/john/.asdf/installs/golang/1.15.2/packages/pkg/mod/goji.io@v2.0.2+incompatible/mux.go:74
net/http.serverHandler.ServeHTTP
        /Users/john/.asdf/installs/golang/1.15.2/go/src/net/http/server.go:2843
net/http.(*conn).serve
        /Users/john/.asdf/installs/golang/1.15.2/go/src/net/http/server.go:1925
```

## Code Review Verification Steps

* [X] Request review from a member of a different team.
* [X] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-5483) for this change
* [Jira subtask](https://dp3.atlassian.net/browse/MB-5543) for this change
* [Jira subtask](https://dp3.atlassian.net/browse/MB-5544) for this change

robot-mymove commented:
<!--
  0 failure: 
  0 warning: 
  1 messages
  
  DangerID: danger-id-Danger;
-->



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>:link: <a href="https://dp3.atlassian.net/browse/MB-5483">MB-5483</a>, <a href="https://dp3.atlassian.net/browse/MB-5543">MB-5543</a>, <a href="https://dp3.atlassian.net/browse/MB-5544">MB-5544</a></td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against b0e229b4e8296f3f02ec80a3d7ac71c3dcbf2615
</p>

pearl-truss reviewed (commented):
I got the expected error but only when I switched the payload to `sendToSyncada: true`. I believe there's just a typo in the description. unless the reverse is expected?

gidjin commented:
> I got the expected error but only when I switched the payload to `sendToSyncada: true`. I believe there's just a typo in the description. unless the reverse is expected?

You're right, I've updated the description.
